google-api-go-generator/gen.go

```diff
@@ -341,7 +347,49 @@ func writeFile(file string, contents []byte) error {
 	if err = os.MkdirAll(outdir, 0755); err != nil {
 		return fmt.Errorf("failed to Mkdir %s: %v", outdir, err)
 	}
-	return os.WriteFile(file, contents, 0644)
+	// Don't try to split spec files, json or non-allowlisted packages
+	if pkg == "" || !apisToSplit[pkg] {
+		return os.WriteFile(file, contents, 0644)
+	}
+
+	// Split generated file out into multiple
+	bs := bytes.Split(contents, []byte(splitFileSeperator))
+	for i, b := range bs {
+		var name string
+		var newB []byte
+		if i == 0 {
+			// For the base case, use the provided inputs as is
+			name = file
+			var err error
+			newB, err = format.Source(b)
+			if err != nil {
+				return err
+			}
+		} else {
+			// determine the new file name
+			base := filepath.Dir(file)
+			fileNum := i + 1
+			name = base + string(filepath.Separator) + pkg + fmt.Sprint(fileNum) + "-gen.go"
```

Comment:
Nit: just making use of std lib a bit more
```suggestion
			name = filepath.Join(base, fmt.Sprintf("%s%d-gen.go", pkg, fileNum))
```

Reply:
done